pkg/syncer/util.go

```diff
@@ -270,3 +275,19 @@ func initVolumeMigrationService(ctx context.Context, metadataSyncer *metadataSyn
 	}
 	return nil
 }
+
+// isPVStaticallyProvisioned checks if the PersistentVolume has been provisioned statically.
+// Dynamically provisioned PVs have a volume attribute called 'storage.kubernetes.io/csiProvisionerIdentity'
+// in their CSI spec, which is set by external-provisioner.
+// Statically provisioned PVs don't have that attribute.
+func isPVStaticallyProvisioned(pv *v1.PersistentVolume) (bool, error) {
+	if pv == nil {
+		return false, errors.New("Expected pv object not to be nil")
+	}
+	if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == csitypes.Name {
+		if _, ok := pv.Spec.CSI.VolumeAttributes[attribCSIProvisionerID]; !ok {
```

Comment:
IMO, we just need to move this check to Line: 894 in pkg/syncer/metadatasyncer.go in this PR.
We do not need this method, since it is performing redundant checks.